rllib/policy/view_requirement.py

```diff
@@ -28,90 +31,91 @@ class ViewRequirement:
+    shift_arr: Optional[np.ndarray] = dataclasses.field(init=False)
+
+    def __post_init__(self):
         """Initializes a ViewRequirement object.
 
-        Args:
-            data_col (Optional[str]): The data column name from the SampleBatch
-                (str key). If None, use the dict key under which this
-                ViewRequirement resides.
-            space (gym.Space): The gym Space used in case we need to pad data
-                in inaccessible areas of the trajectory (t<0 or t>H).
-                Default: Simple box space, e.g. rewards.
-            shift (Union[int, str, List[int]]): Single shift value or
-                list of relative positions to use (relative to the underlying
-                `data_col`).
-                Example: For a view column "prev_actions", you can set
-                `data_col="actions"` and `shift=-1`.
-                Example: For a view column "obs" in an Atari framestacking
-                fashion, you can set `data_col="obs"` and
-                `shift=[-3, -2, -1, 0]`.
-                Example: For the obs input to an attention net, you can specify
-                a range via a str: `shift="-100:0"`, which will pass in
-                the past 100 observations plus the current one.
-            index (Optional[int]): An optional absolute position arg,
-                used e.g. for the location of a requested inference dict within
-                the trajectory. Negative values refer to counting from the end
-                of a trajectory.
-            used_for_compute_actions: Whether the data will be used for
-                creating input_dicts for `Policy.compute_actions()` calls (or
-                `Policy.compute_actions_from_input_dict()`).
-            used_for_training: Whether the data will be used for
-                training. If False, the column will not be copied into the
-                final train batch.
+        shift_arr is infered from the shift value.
+
+        For example:
+            - if shift is -1, then shift_arr is np.array([-1]).
+            - if shift is [-1, -2], then shift_arr is np.array([-2, -1]).
+            - if shift is "-2:2", then shift_arr is np.array([-2, -1, 0, 1, 2]).
         """
-        self.data_col = data_col
-        self.space = (
-            space
-            if space is not None
-            else gym.spaces.Box(float("-inf"), float("inf"), shape=())
-        )
 
-        self.shift = shift
+        if self.space is None:
+            self.space = gym.spaces.Box(float("-inf"), float("inf"), shape=())
 
+        # TODO: ideally we won't need shift_from and shift_to, and shift_step.
+        # all of them should be captured within shift_arr.
         # Special case: Providing a (probably larger) range of indices, e.g.
         # "-100:0" (past 100 timesteps plus current one).
-        self.shift_from = self.shift_to = None
+        self.shift_from = self.shift_to = self.shift_step = None
```

Comment:
for documentation purpose, can we add these cases as unit tests?

Reply:
Added to `test_agent_collector.py`